src/components/StudyRoom/StudyRoomSettings.js

```diff
@@ -0,0 +1,183 @@
+import {PrimaryButton2} from "../CustomMUIComponents/CustomButtons";
+import MockUser from "../Profile/Mocks/mockUser.json"
+import MockStudyRoom from "./Mocks/mockStudyRoom.json"
+
+export default function StudyRoomSettings() {
+    const navigate = useNavigate();
+    const [errorMessage, setErrorMessage] = useState('')
+    const [roomData, setRoomData] = React.useState({
+        title:'',
+        color: '',
+        avatarText:'',
+        description:'',
+        participants:[],
+    });
+
+    const user = MockStudyRoom[1];
+
+    const fetchData = useCallback(() => {
+        axios.get('http://localhost:5000/room/{"username":"test"}')
+            .then(res => {
+                const data = res.data;
+                console.log(JSON.stringify(res.data));
+                setRoomData({...roomData, title: user.title});
+                setRoomData({...roomData, avatarText: user.avatarText});
+                setRoomData({...roomData, color: user.color});
+                setRoomData({...roomData, description: user.description});
+            })
+            .catch(err => {console.log(`Error: ${err}`); setErrorMessage(`${err}`.substring(44) === 401 ? 'request could not be sent' : `${err}`)});
+    },[])
+
+    React.useEffect(()=>{
+        let user = localStorage.getItem("username")
+        setRoomData({...roomData, owner:user});
+        fetchData();
+        console.log(roomData);
+    },[fetchData])
+
+    // sends updated field to db
+    function handleUpdate(e){
+        e.preventDefault();
+        console.log(roomData.title);
+        let avatarIconText = SetAvatarText(roomData.title);
+        console.log('avatar text:', avatarIconText);
+        setRoomData({...roomData, avatarText: avatarIconText});
+        console.log(roomData);
+
+        axios.post('http://localhost:5000/room/',roomData)
+            .then(res => {
+                console.log(res);
+                navigate("/study-room-home");
+            })
+            .catch(err => {console.log(`Error: ${err}`); setErrorMessage(`${err}`.substring(44) === 401 ? 'request could not be sent' : `${err}`)});
+    }
+
+    //deletes the room in db
+    function handleDelete(e){
+
+
+    }
+    function handleTitleChange(e){
+        let avatar = SetAvatarText( e.target.value)
+        setRoomData({...roomData, title: e.target.value, avatarText:avatar});
+        console.log(roomData.title);
+    }
+    function handleColorChange(e){
+        setRoomData({...roomData, color: e.target.value});
+        console.log(roomData.color);
+    }
+    function handleDescriptionChange(e){
+        setRoomData({...roomData, description: e.target.value});
+        console.log(roomData.description);
+        console.log(roomData)
+    }
+
+    function SetAvatarText(t){
+        let finalText='';
+        let displayText = t.split(' ').map(word => word[0]).join('');
+
+        if(displayText.length >4){
+            finalText = displayText.substring(0,4);
+        }else
```

Comment:
missing the { } for the else. If you dont want to use them for single line condition, make it consistent (remove it from the if too)

Reply:
added {} for all if-else statements